client/my-sites/checkout/composite-checkout/hooks/use-stored-cards.ts

```diff
@@ -66,7 +66,11 @@ export default function useStoredCards(
 function storedCardsReducer( state: StoredCardState, action: StoredCardAction ) {
 	switch ( action.type ) {
 		case 'FETCH_END':
-			return { ...state, storedCards: action.payload, isLoading: false };
+			// The endpoint may have returned a non-array without actually failing,
+			// so we need to ensure that storedCards is actually an array here.
+			// eslint-disable-next-line no-case-declarations
+			const storedCards = Array.isArray( action.payload ) ? action.payload : [];
+			return { ...state, storedCards, isLoading: false };
```

Comment:
Do we want to treat this as an error if the data is invalid? 🤔 I guess it depends on the reason that the endpoint did not return an array. I think it might be wise to err on the side of treating it as an error so that the user can see something and might report it rather than just silently ignoring the situation (unless we think this is common). 

To do that we'd need to provide a generic error message and inject it into the state's `error` property here, but that's a little tricky to do because the reducer doesn't have easy access to the translate function since it's not inside a hook or component. ... 🤔  My suggestion is to validate the response data in `useStoredCards` rather than here (keeping the reducer simple). That way we can dispatch `FETCH_ERROR` there with an appropriate message rather than overloading `FETCH_END` here. As a side benefit we also can access `useTranslate` in that hook.